contracts/BIFI/strategies/Arbidex/StrategyArbidexMultiRewardsLP.sol

```diff
@@ -0,0 +1,338 @@
+            uint256[] memory amountOut = IUniswapRouterETH(unirouter).getAmountsOut(arxAvailable, outputToNativeRoute);
+            nativeOut = nativeOut + amountOut[amountOut.length -1];
+        }
+       
+        if (wethAvailable > 0) {
+            nativeOut = nativeOut + wethAvailable;
+        }
+
+        return nativeOut * fees.total / DIVISOR * fees.call / DIVISOR;
+    }
+
+    function setHarvestOnDeposit(bool _harvestOnDeposit) external onlyManager {
+        harvestOnDeposit = _harvestOnDeposit;
+
+        if (harvestOnDeposit) {
+            setWithdrawalFee(0);
+        } else {
+            setWithdrawalFee(10);
+        }
+    }
+
+    function setShouldGasThrottle(bool _shouldGasThrottle) external onlyManager {
+        shouldGasThrottle = _shouldGasThrottle;
+    }
+
+    // called as part of strat migration. Sends all the available funds back to the vault.
+    function retireStrat() external {
+        require(msg.sender == vault, "!vault");
+
+        IMasterChef(chef).emergencyWithdraw(poolId);
+
+        uint256 wantBal = IERC20(want).balanceOf(address(this));
+        IERC20(want).transfer(vault, wantBal);
+    }
+
+    // pauses deposits and withdraws all funds from third party systems.
+    function panic() public onlyManager {
+        pause();
+        IMasterChef(chef).emergencyWithdraw(poolId);
+    }
+
+    function pause() public onlyManager {
+        _pause();
+
+        _removeAllowances();
+    }
+
+    function unpause() external onlyManager {
+        _unpause();
+
+        _giveAllowances();
+
+        deposit();
+    }
+
+    function _giveAllowances() internal {
+        IERC20(want).safeApprove(chef, type(uint).max);
+        IERC20(output).safeApprove(unirouter, type(uint).max);
+        IERC20(native).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken0).safeApprove(unirouter, 0);
+        IERC20(lpToken0).safeApprove(unirouter, type(uint).max);
+
+        IERC20(lpToken1).safeApprove(unirouter, 0);
+        IERC20(lpToken1).safeApprove(unirouter, type(uint).max);
+
+        if (rewardToNativeRoute.length != 0) {
+            for (uint i; i < rewardToNativeRoute.length; i++) {
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, 0);
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, type(uint).max);
+            }
+        }
+    }
+
+    function _removeAllowances() internal {
+        IERC20(want).safeApprove(chef, 0);
+        IERC20(output).safeApprove(unirouter, 0);
+        IERC20(native).safeApprove(unirouter, 0);
+
+        IERC20(lpToken0).safeApprove(unirouter, 0);
+        IERC20(lpToken1).safeApprove(unirouter, 0);
+
+        if (rewardToNativeRoute.length != 0) {
+            for (uint i; i < rewardToNativeRoute.length; i++) {
+                IERC20(rewardToNativeRoute[i][0]).safeApprove(unirouter, 0);
+            }
+        }
+    }
+
+    function addRewardRoute(address[] memory _rewardToNativeRoute) external onlyOwner {
```

Comment:
i would add require here that reward[0] != want and maybe not native

Reply:
Done